### PR TITLE
refac: userEvent Setup Code

### PR DESCRIPTION
**Pull request by ssi02014**
## Overview

Starting with userEvent `v14`, it is recommended to call `userEvent.setup()` before rendering the component.

I have refactored the code using userEvent as per the guide.

https://testing-library.com/docs/user-event/intro#writing-tests-with-userevent

## PR Checklist

- [x] I read and included theses actions below

1. I have read the [Contributing Guide](https://github.com/toss/slash/blob/main/.github/CONTRIBUTING.md)
2. I have written documents and tests, if needed.

### CODE REVIEW

File: packages/react/error-boundary/src/ErrorBoundaryGroup.spec.tsx [hunk 1]
```diff
@@ -45,8 +46,6 @@ describe('ErrorBoundaryGroup', () => {
 
     expect(screen.getByText(`An error occurred: This is an error`)).toBeInTheDocument();
 
-    const user = await userEvent.setup();
```
Comment (ssi02014) [inline packages/react/error-boundary/src/ErrorBoundaryGroup.spec.tsx:48]:
```ts
setup: (options?: Options | undefined) => UserEvent
```
The return value of userEvent.setup is not a Promise.

**Review by okinawaa — APPROVED** [review]
I agree with this PR

File: packages/react/react/src/hooks/useOutsideClickEffect.spec.tsx [hunk 2]
```diff
@@ -42,13 +42,14 @@ describe('useOutsideClickEffect', () => {
   }
 
   it('컨테이너 바깥에 위치한 DOM에서 이벤트가 발생하면 콜백이 호출된다.', async () => {
+    const user = userEvent.setup();
     const onEffect = jest.fn();
     prepare({ onEffect });
 
-    userEvent.click(screen.getByTestId('container-0'));
```
Comment (ssi02014) [inline packages/react/react/src/hooks/useOutsideClickEffect.spec.tsx:48]: Starting with v14, it is recommended to add "await" to multiple guides.

### reference
https://github.com/testing-library/user-event/discussions/1050
https://testing-library.com/docs/user-event/convenience

**Comment by ssi02014** [comment]
@okinawaa thanks!! 🤗🎉

**Pull request merged**